pkg/blobstore/local/periodic_syncer.go

```diff
@@ -144,13 +156,25 @@ func (ps *PeriodicSyncer) ProcessBlockPut() {
 		// The system was idle for some time. Wait a bit, so
 		// that the current epoch gets a meaningful amount of
 		// data.
-		<-ch
-		_, t = ps.clock.NewTimer(ps.minimumEpochInterval)
+		select {
+		case <-ctx.Done():
+			// t can stay nil,
+			break
+		case <-ch:
+			_, t = ps.clock.NewTimer(ps.minimumEpochInterval)
+		}
+	}
+	var isFinalSync bool
+	select {
+	case <-ctx.Done():
+		ps.lastSynchronizationTime = ps.clock.Now()
+		isFinalSync = true
```

Comment:
Do we think it's necessary to do two syncs in this case?
- Once with `isFinalSync` set to `false`,
- Another one with `isFinalSync` set to `true`.

That way we first do a 'catch-up', where we try to flush as much data as possible. Only once that's done, there will be a brief point in time in which the service doesn't accept writes.

Reply:
Wired it up with two syncs this time.